File: src/main/java/pe/edu/upeu/turismospringboot/model/dto/ChatResumenDto.java
package pe.edu.upeu.turismospringboot.model.dto;

import lombok.Data;
import pe.edu.upeu.turismospringboot.model.enums.EstadoMensaje;

import java.time.LocalDateTime;

@Data
public class ChatResumenDto {
    private String username;
    private String nombreCompleto;
    private String ultimoMensaje;
    private LocalDateTime hora;
    private String avatarUrl;
    private EstadoMensaje estadoUltimoMensaje;
}

File: turismo_flutter/lib/features/admin/data/models/chat_resumen_dto.dart
class ChatResumenDto {
  String? username;
  String? nombreCompleto;
  String? ultimoMensaje;
  DateTime? hora;
  String? avatarUrl;
  String? estadoUltimoMensaje;

  ChatResumenDto({
    this.username,
    this.nombreCompleto,
    this.ultimoMensaje,
    this.hora,
    this.avatarUrl,
    this.estadoUltimoMensaje,
  });

  factory ChatResumenDto.fromJson(Map<String, dynamic> json) {
    return ChatResumenDto(
      username: json['username'],
      nombreCompleto: json['nombreCompleto'],
      ultimoMensaje: json['ultimoMensaje'],
      hora: json['hora'] != null ? DateTime.parse(json['hora']) : null,
      avatarUrl: json['avatarUrl'],
      estadoUltimoMensaje: json['estadoUltimoMensaje'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'nombreCompleto': nombreCompleto,
      'ultimoMensaje': ultimoMensaje,
      'hora': hora?.toIso8601String(),
      'avatarUrl': avatarUrl,
      'estadoUltimoMensaje': estadoUltimoMensaje,
    };
  }
}